pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,105 @@
+graph TD
+    PR[Producer]-- Message --- EG[AsyncAPI Event-Gateway]
+    EG -- Message --- EGV{Is Message valid?}
+    EGV -->|Yes| BR[Broker]
+    EGV --- |No| INV{Fail when invalid?}
+    INV -->|Yes| ERR[/Fail/] -- Produce request errored --> PR
+    INV -->|No| BR
+```
+The AsyncAPI document is important because payload schemas are taken from it, and messages are validated against it in your application.
+You can spin up the AsyncAPI gateway using an AsyncAPI file. All the messages are forwarded to a WebSocket endpoint; if the message/payload is invalid, it includes a validation error message.
+
+### Schema Registry validation
+Producers and consumers do not communicate with each other directly; rather, information transfer happens via Kafka. At the same time, the consumer still needs to know the type of data the producer is sending. Imagine if the producer starts sending bad data to Kafka or if the data type of your data gets changed. We need a way to have a common data type that must be agreed upon.
+
+This is where Schema Registry comes into play. It is an application that runs outside your Kafka protocol and handles schema distribution to producers and consumers by storing a copy of the schema in its local cache and validating them in Kafka.
+
+```mermaid
+sequenceDiagram
+Producer ->> Schema Registry: Register/checks message schema
+Schema Registry ->> Producer: Return registration result
+
+Producer ->> Kafka: Publish message
+Kafka ->> Consumer: Recieves message
+Consumer ->> Schema Registry: Validate message schema
+Schema Registry ->> Consumer: Return validation result
+```
+
+With the Schema Registry in place, the producer first talks to the Schema Registry and checks if the schema of the message it wants to send is available before sending it to the broker. If it cannot locate the schema, it registers it in the Schema Registry. Then the producer sends a message to the broker prefixed with a unique schema ID. When the consumer processes this message, it will communicate with the Schema Registry using the schema ID obtained from the producer. If there is a schema mismatch, the Schema Registry will throw an error, informing the producer that it violates the schema agreement.
+
+AsyncAPI is not directly involved in validation based on the Schema Registry. The good thing is that you do not have to duplicate schemas in your AsyncAPI document stored in Schema Registry. You can reference schemas from Schema Registry in your AsyncAPI documents.
+Here's an example of an AsyncAPI document where you can see both `schemaFormat` and `payload` referenced from the Schema Registry:
+```yml
+asyncapi: 3.0.0
+info:
+  title: Example with Avro
+  version: 0.1.0
+
+channels:
+  example:
+    address: 'example'
+    messages:
+      avroMessage:
+        payload:
+          schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
+          schema:
+            $ref: 'https://example.europe-west3.gcp.confluent.cloud/subjects/test/versions/1/schema'
```

Comment:
```suggestion
asyncapi: 3.0.0
info:
  title: Example with Avro
  version: 0.1.0

channels:
  example:
    address: 'example'
    messages:
      avroMessage:
        payload:
          schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
          schema:
            $ref: 'https://raw.githubusercontent.com/asyncapi/website/20a31a0396b41dd24b1bac877ab7ce3f58037c28/public/resources/casestudies/adeo/CostingRequestPayload.avsc'

operations:
  onMessage:
    action: receive
    channel:
      $ref: '#/channels/example'
```

all would be good but after your change, the example was missing an operation so my code snipped:
- add operation
- update the schema link to a link that actually works and anyone can render this document in studio

Reply:
you do not have to do anything on local, if you agree just click commit suggestion
![Screenshot 2023-11-08 at 17 12 34](https://github.com/asyncapi/website/assets/6995927/45ed3758-d24e-40ca-abe1-8fe850b65b55)

Reply:
> you do not have to do anything on local, if you agree just click commit suggestion ![Screenshot 2023-11-08 at 17 12 34](https://user-images.githubusercontent.com/6995927/281469471-45ed3758-d24e-40ca-abe1-8fe850b65b55.png)

Thank you for the clarification.

Reply:
resolved with bac086f